File: Services_Play_Music_10Aug21/app/src/main/java/com/example/servicesplaymusic10aug21/MusicService.java
package com.example.servicesplaymusic10aug21;

import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Build;
import android.os.IBinder;
import android.view.LayoutInflater;
import android.view.View;

import androidx.constraintlayout.widget.ConstraintSet;
import androidx.core.app.NotificationCompat;

import java.util.Objects;

public class MusicService extends Service {
    private MediaPlayer mediaPlayer;

    public MusicService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mediaPlayer = MediaPlayer.create(this, R.raw.ride_it);
        showNotificationAndStartForeGround();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        mediaPlayer.start();
        return super.onStartCommand(intent, flags, startId);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mediaPlayer.stop();
    }

    private void showNotificationAndStartForeGround() {
        createChannel();
        View view = LayoutInflater.from(this).inflate(R.layout.notification_view, , false);
        NotificationCompat.Builder notificationBuilder = null;
        notificationBuilder = new NotificationCompat.Builder(this, "CHANNEL_ID")
                .setContentTitle("Music Playing..")
                .setContent()
                .setPriority(NotificationCompat.PRIORITY_HIGH);

        Notification notification = null;
        notification = notificationBuilder.build();
        startForeground(1, notification);
    }



    /*
Create notification channel if OS version is greater than or equal to Oreo
*/
    public void createChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel("CHANNEL_ID", "CHANNEL_NAME", NotificationManager.IMPORTANCE_DEFAULT);
            channel.setDescription("Music Notifications");
            Objects.requireNonNull(this.getSystemService(NotificationManager.class)).createNotificationChannel(channel);
        }
    }
}

File: _Kotlin Projects/Unit_3_Coding_Evalution_2/app/src/main/java/com/example/unit_3_coding_evalution_2/MusicService.kt
package com.example.unit_3_coding_evalution_2

import android.app.*
import android.content.Intent
import android.media.MediaPlayer
import android.net.Uri
import android.os.Build
import android.os.IBinder
import android.view.LayoutInflater
import androidx.core.app.NotificationCompat

class MusicService : Service() {
    private lateinit var mediaPlayer: MediaPlayer
    private lateinit var url: String
    private val channelId = "Music Playing Notification"

    override fun onBind(intent: Intent): IBinder {
        TODO("Return the communication channel to the service.")
    }

    override fun onCreate() {
        super.onCreate()
        showNotificationAndStartForeGround()
        mediaPlayer = MediaPlayer()
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        if (intent != null) {
            url = intent.getStringExtra("song").toString()
        }
        playMusic()
        return super.onStartCommand(intent, flags, startId)

    }

    private fun playMusic() {
        mediaPlayer.reset()
        val uri = Uri.parse(url)
        mediaPlayer = MediaPlayer.create(applicationContext, uri)
        mediaPlayer.start()
    }

    private fun showNotificationAndStartForeGround(): Int {
        createChannel();
        val notificationIntent = Intent(this, MusicPlayActivity::class.java)
        val pendingIntent = PendingIntent.getActivity(this, 0, notificationIntent, 0)
        val notification: Notification = NotificationCompat.Builder(this, channelId)
            .setContentTitle("Music Playing")
            .setContentText("Music Playing")
            .setContentIntent(pendingIntent)
            .build()
        startForeground(1, notification)
        return START_NOT_STICKY
    }


    override fun onDestroy() {
        super.onDestroy()
        mediaPlayer.stop()
    }

    private fun createChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val channel = NotificationChannel(
                channelId,
                "Music Playing",
                NotificationManager.IMPORTANCE_DEFAULT
            );
            channel.description = "Music Notifications";
            (getSystemService(NOTIFICATION_SERVICE) as NotificationManager).createNotificationChannel(
                channel
            );
        }
    }
}